### ISSUE TITLE
Intra link bug, link name being recognized as link

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust#51452]
Example:

```rust
/// [ARM's NEON Intrinsics Online Database][arm_dat] is also useful.
///
/// [arm_dat]: https://developer.arm.com/technologies/neon/intrinsics
```

We get:

```
warning: `[arm_dat]` cannot be resolved, ignoring it...
 --> libcore/../stdsimd/coresimd/aarch64/mod.rs:4:45
  |
1 | /// [ARM's NEON Intrinsics Online Database][arm_dat] is also useful.
  |                                             ^^^^^^^ cannot be resolved, ignoring

warning: `[arm_dat]` cannot be resolved, ignoring it...
 --> libcore/../stdsimd/coresimd/aarch64/mod.rs:9:6
  |
3 | /// [arm_dat]: https://developer.arm.com/technologies/neon/intrinsics
  |      ^^^^^^^ cannot be resolved, ignoring
```

**Comment by kinnison** [comment]
I had a brief attempt at reproducing this with the current compiler and couldn't.  I created a rustdoc with: `./x.py doc -i --stage 1 src/libstd` and then ran it against a `test.rs` containing the example from the initial post, with `///` converted to `//!` as a trivial test case.  I ran it with `.../rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustdoc --edition 2018 -v test.rs` It did not warn and did correctly generate the docs.

If someone knows how to successfully reproduce the issue I'd be interested in taking a look if that'd help.

**Comment by rossmacarthur** [comment]
Considering that this issue is not reproducible, and is so old. I think this issue should be closed. 

@steveklabnik ?

**Comment by steveklabnik** [comment]
Yeah, sounds good. @GuillaumeGomez let me know if you're still seeing this, but so much has changed since you filed the bug I bet it's been fixed.
